doc/Makefile

```
@@ -7,29 +7,19 @@ SPHINXAUTOGEN = sphinx-autogen
 BUILDDIR      = _build
 
 # Internal variables.
-ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees  $(SPHINXOPTS) .
+ALLSPHINXOPTS = -d $(BUILDDIR)/doctrees  $(SPHINXOPTS) .
 
 .PHONY: help clean html linkcheck doctest api
 
-all: html
-
 help:
 	@echo "Please use \`make <target>' where <target> is one of"
 	@echo "  all        generate the complete webpage"
-	@echo "  html       make only the HTML files from the existing rst sources"
-	@echo "  linkcheck  check all external links for integrity"
-	@echo "  doctest    run all doctests embedded in the documentation (if enabled)"
+	@echo "  html       build the HTML files from the existing rst sources"
+	@echo "  api        generate rst source files of API documentation"
+	@echo "  server     make a local HTTP server for previewing the built documentation"
+	@echo "  clean      clean up built and generated files"
```

Maybe sort these lines alphabetically?
```suggestion
	@echo "  api        generate rst source files of API documentation"
	@echo "  clean      clean up built and generated files"
	@echo "  html       build the HTML files from the existing rst sources"
	@echo "  server     make a local HTTP server for previewing the built documentation"
```

I'd also prefer it if the rest of this Makefile was sorted alphabetically (i.e. all, api, clean, html, server), while keeping this `help` target on top of course.

---

People usually put `clean` to be the last one, while the target like `all` or `install`/`build`/`html` is the first one in a makefile. Shall we keep this rule? @seisman @willschlitzer How do you think?

---

I prefer to put `clean` at the bottom.

---

@weiji14 @seisman Change the order to be `all`, `api`, `html`, `server`, and `clean` in e719f74. So keep `all` first, `clean` last, and sort the other alphabetically.